src/transformers/pipelines/base.py

```diff
@@ -1010,6 +1010,26 @@ def __init__(
             self.device = self.model.device
         logger.warning(f"Device set to use {self.device}")
 
+        # Ensure the dtype chosen (possibly coming from `torch_dtype="auto"`) is actually supported on the
+        # execution device. If it is not (e.g. the model weights were loaded as bfloat16 but the target device
+        # does not have bfloat16 capabilities), silently fall back to FP32 so that users still get a working
+        # pipeline instead of a runtime error. The original dtype is preserved when possible.
+        if is_torch_available() and self.framework == "pt":
+            model_dtype = getattr(self.model, "dtype", None)
+            if model_dtype is not None and isinstance(self.device, torch.device):
+                try:
+                    # Allocation will fail fast if the dtype is not supported on the target device.
+                    _ = torch.empty(1, dtype=model_dtype, device=self.device)
+                except (TypeError, RuntimeError):
+                    logger.warning(
+                        f"torch_dtype={model_dtype} is not supported on device {self.device}. "
+                        "Falling back to torch.float32."
+                    )
+                    # Cast the model weights to fp32 and update the config so subsequent saves are correct.
+                    self.model = self.model.to(dtype=torch.float32)
+                    if hasattr(self.model, "config") and hasattr(self.model.config, "torch_dtype"):
+                        self.model.config.torch_dtype = torch.float32
```

Comment:
(I couldn't find a reliable source to answer my assumption below, so I might be wrong!)

This assumes we can always load a model into a device with `from_pretrained` using the `torch_dtype` in the `config.json`, even if the device doesn't support the dtype. I would be surprised if this is possible 🤔 If it is not possible, a potential solution would be a try/except at model loading time (`infer_framework_load_model` in this class): try loading with `auto`, fall back to `fp32`